src/guidellm/backend/openai.py

```diff
@@ -1,15 +1,16 @@
-import functools
-from typing import Any, Dict, Generator, List, Optional
+from typing import TYPE_CHECKING, Any, Dict, Generator, List, Optional
 
+from config import settings
 from loguru import logger
 from openai import OpenAI, Stream
-from openai.types import Completion
 from transformers import AutoTokenizer
 
-from config import settings
 from guidellm.backend import Backend, BackendEngine, GenerativeResponse
 from guidellm.core import TextGenerationRequest
 
+if TYPE_CHECKING:
+    from openai.types import Completion
```

Comment:
Do we really need it?
I would suggest importing it as usual, so we can detect circular imports faster. And they point to architecture problems in the code structure. might be helpful i believe.

Reply:
Since these pathways are for third-party libraries, we shouldn't be able to hit any circular imports there. It was recommended to add a bit more performance for the pyc files, and based on the spec, seemed like a reasonable thing to do. I'm indifferent on fully enforcing it, but see the spec here for more background: https://docs.astral.sh/ruff/rules/typing-only-third-party-import/

Reply:
Well.

> Unused imports add a performance overhead at runtime

I believe that the impact of such operations is tiny. We don't use any 3-rd party imports that have any sort of computation. I mean calling it overhead is quite ridiculous 😄. Like we use Python, common.

> and risk creating import cycles

So this is the topic that I am talking about. If you have the cyclic import in your code - this is literally the architecture problem.  Like don't you think that the regular import is going to be way easier because of the caching than having such an `if` statement in most of the modules on a distance?

_P.S. I was using the `typing.TYPE_CHECKING` before but seems it gives nothing if you don't **_check types_** and import modules that execute the code on import which is usually wild for most of the packages._